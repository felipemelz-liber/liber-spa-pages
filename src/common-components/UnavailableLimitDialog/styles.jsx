import styled from 'styled-components';
import { FlatButton } from 'liber-components';

const TEXT_COLOR = '#6287a7';

export const Container = styled.div`
  padding: 24px 24px 12px 24px;
  width: 480px;

  * {
    font-family: Roboto;
  }
`;

export const Header = styled.div`
  font-size: 21px;
  font-weight: bold;
  line-height: 1.4;
  letter-spacing: 0.08px;
  color: ${TEXT_COLOR};
  margin-bottom: 24px;
`;

export const Content = styled.div`
  font-size: 16px;
  line-height: 1.38;
  letter-spacing: 0.01px;
  margin-bottom: 24px;
  color: ${TEXT_COLOR};
  margin-bottom: 24px;
`;

export const ActionsRow = styled.div`
  display: flex;
  justify-content: flex-end;
`;

export const Button = styled(FlatButton).attrs(() => ({ size: 'large' }))``;
