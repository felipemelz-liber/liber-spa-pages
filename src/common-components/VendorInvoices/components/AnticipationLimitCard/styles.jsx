import styled from 'styled-components';

export const Card = styled.div`
  font-family: Roboto;
  color: #6287a7;
  width: 360px;
  box-shadow: 0px 3px 6px #00000029;
  border: 1px solid #cdd7df;
  border-radius: 8px;
  padding: 16px 24px;
  margin-bottom: 56px;

  * {
    font-family: Roboto;
  }
`;

export const CardTitle = styled.div`
  font-size: 16px;
  line-height: 22px;
  letter-spacing: 0.06px;
  font-weight: 500;
  color: #4b6f85;
  margin-bottom: 4px;
`;
