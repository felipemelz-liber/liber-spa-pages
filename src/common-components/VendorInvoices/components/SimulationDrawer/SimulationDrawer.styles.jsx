import styled, { css } from 'styled-components';
import { Button } from 'liber-components';

export const Title = styled.span`
  font-family: Roboto;
  font-size: 20px;
  font-weight: 500;
  font-stretch: normal;
  font-style: normal;
  line-height: 1.4;
  letter-spacing: normal;
  text-align: left;
  color: #4b6f85;
`;

export const SelectedNumber = styled(Title)`
  font-weight: normal;
`;

export const ConfirmationButton = styled(Button)`
  background-color: #002874;
  --button-ripple-color: #002d80;
  --button-text-color: #fafdff;
  min-width: 235px;
  &:hover {
    background-color: #001f59;
  }
`;

export const Row = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  padding: 16px;
  ${({ hasPaddingTop }) =>
    hasPaddingTop
      ? null
      : css`
          padding-top: 0px;
        `}
  align-items: center;
  & > div {
    margin-top: 0px;
  }
`;
