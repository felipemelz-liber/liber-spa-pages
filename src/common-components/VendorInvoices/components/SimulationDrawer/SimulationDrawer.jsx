/* eslint-disable max-len */
import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Icon from '@mdi/react';
import { mdiFinance } from '@mdi/js';
import { Drawer, Button } from 'liber-components';
import { Title, SelectedNumber, Row, ConfirmationButton } from './SimulationDrawer.styles';
import { SimulationInfo } from '../../../../views/fornecedor-duplicatas/components/SimulationDrawer/components/SimulationInfo/SimulationInfo';
import { SimulationDialog } from '../../../../views/fornecedor-duplicatas/components/SimulationDrawer/components/SimulationDialog/SimulationDialog';
import SimulationTerms from '../../../../views/fornecedor-duplicatas/components/SimulationDrawer/components/SimulationTerms/SimulationTerms';
import LoadingButton from '../../../Buttons/LoadingButton';
import {
  requestSimulationReport,
  anticipateInvoices,
  calculateQuote,
} from '../../actions/VendorInvoicesActions';
import { dispatchDataLayerEvent } from '../../../../vendor/Utils';

export const SimulationDrawer = ({
  selectedInvoices,
  quote,
  rateByPeriod,
  requestReport,
  requestAnticipation,
  changeQuote,
  loading,
  isSimulationOutdated,
}) => {
  React.useEffect(() => {
    if (selectedInvoices.length > 0) {
      changeQuote();
    }
  }, [selectedInvoices]);

  const [open, setOpen] = useState(false);
  const [canDownload, setCanDownload] = useState(false);

  const handleConfirmation = () => {
    dispatchDataLayerEvent('titulosBaixarRelatorioDaSimulacao');

    setOpen(false);
    setCanDownload(false);
    requestReport();
  };

  return (
    <>
      <Drawer
        blockNavigation={false}
        position="bottom"
        drawerSize={1120}
        size={128}
        show={selectedInvoices.length > 0}
      >
        <Row hasPaddingTop>
          <Title>
            Simulação de Adiantamento{' '}
            <SelectedNumber>{`(${selectedInvoices.length} títulos)`}</SelectedNumber>
          </Title>
          <Button version={2} size="small" onClick={() => setOpen(true)}>
            <Icon path={mdiFinance} />
            Baixar Relatório da Simulação (Excel)
          </Button>
        </Row>
        <Row>
          <SimulationInfo
            {...quote}
            selectedInvoices={selectedInvoices}
            rateByPeriod={rateByPeriod}
            isSimulationOutdated={isSimulationOutdated}
          >
            <LoadingButton
              onClick={requestAnticipation}
              loading={loading && !isSimulationOutdated}
              ButtonComponent={ConfirmationButton}
              size="large"
              disabled={isSimulationOutdated}
            >
              SOLICITAR ADIANTAMENTO
            </LoadingButton>
          </SimulationInfo>
        </Row>
      </Drawer>
      <SimulationDialog
        title="Termos do Relatório de Simulação"
        open={open}
        onClose={() => {
          setOpen(false);
          setCanDownload(false);
        }}
        maxWidth={592}
        confirmationButtonProps={{
          text: 'CONTINUAR',
          onConfirmation: handleConfirmation,
          disabled: !canDownload,
        }}
      >
        <SimulationTerms checked={canDownload} onChange={() => setCanDownload(!canDownload)} />
      </SimulationDialog>
    </>
  );
};

SimulationDrawer.propTypes = {
  selectedInvoices: PropTypes.arrayOf(PropTypes.number),
  quote: PropTypes.shape({
    faceValue: PropTypes.number,
    discountValue: PropTypes.number,
    discountRate: PropTypes.number,
    netValue: PropTypes.number,
  }),
  rateByPeriod: PropTypes.bool,
  requestReport: PropTypes.func,
  requestAnticipation: PropTypes.func,
  changeQuote: PropTypes.func,
  loading: PropTypes.bool,
  isAdmin: PropTypes.bool,
  isSimulationOutdated: PropTypes.bool,
};

SimulationDrawer.defaultProps = {
  selectedInvoices: [],
  quote: {},
  rateByPeriod: false,
  requestReport: () => {},
  requestAnticipation: () => {},
  changeQuote: () => {},
  loading: false,
  isAdmin: false,
  isSimulationOutdated: false,
};

const mapStateToProps = ({
  vendorInvoices: { selectedInvoices, quote, rateByPeriod, loading, isAdmin, isSimulationOutdated },
}) => ({
  selectedInvoices,
  quote,
  rateByPeriod,
  loading,
  isAdmin,
  isSimulationOutdated,
});

const mapDispatchToProps = {
  requestReport: requestSimulationReport,
  requestAnticipation: anticipateInvoices,
  changeQuote: calculateQuote,
};

export default connect(mapStateToProps, mapDispatchToProps)(SimulationDrawer);
