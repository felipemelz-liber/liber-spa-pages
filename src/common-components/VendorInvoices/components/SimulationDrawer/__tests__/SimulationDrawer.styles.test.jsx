import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import * as styles from '../SimulationDrawer.styles';

describe('SimulationDrawer styled components test', () => {
  configure({ adapter: new Adapter() });
  const { Row, ...Components } = styles;

  Object.keys(Components).forEach(componentName => {
    const Component = Components[componentName];
    it(`should render ${componentName} correctly`, () => {
      const wrapper = shallow(<Component />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  it('should render Row correctly', () => {
    const wrapper = shallow(<Row />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render Row with padding top correctly', () => {
    const wrapper = shallow(<Row hasPaddingTop />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
