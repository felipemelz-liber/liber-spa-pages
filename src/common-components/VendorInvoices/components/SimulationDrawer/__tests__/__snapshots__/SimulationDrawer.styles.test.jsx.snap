// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimulationDrawer styled components test should render ConfirmationButton correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-htpNat",
        "isStatic": false,
        "rules": Array [
          "
  background-color: #002874;
  --button-ripple-color: #002d80;
  --button-text-color: #fafdff;
  min-width: 235px;
  &:hover {
    background-color: #001f59;
  }
",
        ],
      },
      "displayName": "Styled(Buttons__Button)",
      "foldedComponentIds": Array [
        "Buttons__Button-u35aut-0",
      ],
      "render": [Function],
      "styledComponentId": "sc-htpNat",
      "target": Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      },
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`SimulationDrawer styled components test should render Row correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bxivhb",
        "isStatic": false,
        "rules": Array [
          "
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  padding: 16px;
  ",
          [Function],
          "
  align-items: center;
  & > div {
    margin-top: 0px;
  }
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bxivhb",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`SimulationDrawer styled components test should render Row with padding top correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bxivhb",
        "isStatic": false,
        "rules": Array [
          "
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  padding: 16px;
  ",
          [Function],
          "
  align-items: center;
  & > div {
    margin-top: 0px;
  }
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bxivhb",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  hasPaddingTop={true}
/>
`;

exports[`SimulationDrawer styled components test should render SelectedNumber correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bwzfXH",
        "isStatic": false,
        "rules": Array [
          "
  font-family: Roboto;
  font-size: 20px;
  font-weight: 500;
  font-stretch: normal;
  font-style: normal;
  line-height: 1.4;
  letter-spacing: normal;
  text-align: left;
  color: #4b6f85;
",
          "
  font-weight: normal;
",
        ],
      },
      "displayName": "Styled(styled.span)",
      "foldedComponentIds": Array [
        "sc-bdVaJa",
      ],
      "render": [Function],
      "styledComponentId": "sc-bwzfXH",
      "target": "span",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`SimulationDrawer styled components test should render Title correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  font-family: Roboto;
  font-size: 20px;
  font-weight: 500;
  font-stretch: normal;
  font-style: normal;
  line-height: 1.4;
  letter-spacing: normal;
  text-align: left;
  color: #4b6f85;
",
        ],
      },
      "displayName": "styled.span",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "span",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
